src/dataLoaderService.ts

```diff
@@ -0,0 +1,56 @@
+import DataLoader, { BatchLoadFn } from 'dataloader';
+import { Context } from './Context';
+import { getShortUrl } from './common';
+
+export class DataLoaderService {
+  protected loaders: Record<string, DataLoader<unknown, unknown>>;
+  private ctx: Context;
+
+  constructor({ ctx }: { ctx: Context }) {
+    this.loaders = {};
+    this.ctx = ctx;
+  }
+
+  protected getLoader<K, V>({
```

Comment:
This is a very specific implementation but that's ok for now. I don't expect to have too many dataloaders. Thankfully graphorm solves a lot of issues

Reply:
Yeah, I think graphorm works really well for anything db related but when you have 3party service calls (like shortener) it becomes inefficient.

For my knowledge, I would be Interested in what terms you think this is specific implementation? We can also catch up on a short call outside of this to discuss. I would appreciate it.